internal/provider/generate.go

```diff
@@ -498,13 +506,31 @@ provider %[1]q {
 		return nil, err
 	}
 
-	g.infof("getting Terraform binary")
-	installer := &releases.ExactVersion{
-		Product:    product.Terraform,
-		Version:    version.Must(version.NewVersion("1.0.5")),
-		InstallDir: tmpDir,
+	i := install.NewInstaller()
+	sources := []src.Source{}
+	if g.tfVersion != "" {
+		g.infof("download specific Terraform binary version %s", g.tfVersion)
```

Comment:
Since these log messages are intended to let the human operator know about what the command is about to do, I think this should be something along the lines of:

```suggestion
		g.infof("downloading Terraform CLI binary version from releases.hashicorp.com: %s", g.tfVersion)
```